File: 2540-minimum-common-value/2540-minimum-common-value.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        result = int(1e12)
        one, two = 0, 0

        while one < len(nums1) and two < len(nums2):
            if nums1[one] == nums2[two]:
                result = nums1[one]
                break
            elif nums1[one] < nums2[two]:
                one += 1
            else:
                two += 1

        return -1 if result == int(1e12) else result

File: 2540-minimum-common-value/2540-minimum-common-value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        int result = Integer.MAX_VALUE;
        int one = 0, two = 0;

        while (one < nums1.length && two < nums2.length) {
            if (nums1[one] == nums2[two]) {
                result = nums1[one];
                break;
            } else if (nums1[one] < nums2[two]) {
                one++;
            } else {
                two++;
            }
        }

        return result == Integer.MAX_VALUE ? -1 : result;
    }
}